route_guide/route_guide_server.cc

```diff
@@ -55,133 +64,235 @@ float GetDistance(const Point& start, const Point& end) {
+  float delta_lon_rad = ConvertToRadians(lon_2 - lon_1);
 
-  float a = pow(sin(delta_lat_rad/2), 2) + cos(lat_rad_1) * cos(lat_rad_2) *
-            pow(sin(delta_lon_rad/2), 2);
-  float c = 2 * atan2(sqrt(a), sqrt(1-a));
+  float a = pow(sin(delta_lat_rad / 2), 2)
+      + cos(lat_rad_1) * cos(lat_rad_2) * pow(sin(delta_lon_rad / 2), 2);
+  float c = 2 * atan2(sqrt(a), sqrt(1 - a));
   int R = 6371000; // metres
 
   return R * c;
 }
 
-std::string GetFeatureName(const Point& point,
-                           const std::vector<Feature>& feature_list) {
+// Just return the name of feature in feature_list vector
+std::string GetFeatureName(
+    const Point& point,
+    const std::vector<Feature>& feature_list) {
   for (const Feature& f : feature_list) {
-    if (f.location().latitude() == point.latitude() &&
-        f.location().longitude() == point.longitude()) {
+    if (
+        f.location().latitude() == point.latitude()
+        && f.location().longitude() == point.longitude()) {
       return f.name();
     }
   }
   return "";
 }
 
+
+// The idea of the RouteGuideImpl class is based on the stout-grpc interface
+// which uses callbacks. stout-grpc is intented to be a higher-level interface
+// while still being asynchronous that uses stout-eventuals
+// (https://github.com/3rdparty/stout-eventuals) to make composing async code
+// easier.
+// This class contains 4 methods: ServeGetFeature(unary rpc), ServeListFeatures
+// (server streaming), ServeRecordRoute(client streaming), ServeRouteChat
+// (bidirectional streaming).
 class RouteGuideImpl final {
  public:
+  // Populate feature_list_ vector with addresses from database
+  // (route_guide_db.json)
   explicit RouteGuideImpl(const std::string& db) {
     routeguide::ParseDb(db, &feature_list_);
   }
 
-  auto Serve(Server* server) {
-    auto status = server->Serve<RouteGuide, Point, Feature>(
-        "GetFeature",
-        [this](auto* call, auto&& point) {
-          Feature feature;
-          feature.set_name(GetFeatureName(*point, feature_list_));
-          feature.mutable_location()->CopyFrom(*point);
-          call->WriteAndFinish(feature, Status::OK);
-        },
-        [](auto*, bool) {});
-
-    if (!status.ok()) {
-      return status;
-    }
+  // The goal of this function is to serve unary rpc call. Client sends to the
+  // server the request which contains Point's coordinates (longitude,
+  // latitude). The server checks received Point's coordinates, if database has
+  // the specific address which corresponds to Point's coordinates - the server
+  // will response with it address.
+  // Accept<...>() method (stout/grpc/server.h) gets the name of rpc call and
+  // the host name and returns the Composable structure composed with Map()
+  // using overloaded operator "|". Then we compose with Map() in order to
+  // have deal with multiple connections and finally compose with Loop() to
+  // convert stream into the eventual.
```

Comment:
Some clarifying textual suggestions here too:
```suggestion
  // ServeGetFeature serves unary (one-request, one-response) rpc calls. 
  // The client sends the server a request containing a Point (latitude, longitude).
  // If the server knows the address that corresponds to the Point's coordinates
  // it will response with that address.
  //
  // The `Accept<...>()` method (stout/grpc/server.h) gets the name of the
  // RPC function, and creates an "Eventual" that will produce a stream of 
  // connections. Eventuals can be composed with other Eventuals using
  // the "|" operator, and here we compose `Accept` with a subsequent 
  // `Map` to handle each connection in the stream with a `Server::Handler()`.
  // We end the chain of Eventuals processing each connection by composing 
  // with `Loop`, which takes us back to the top of the streaming chain to
  // process the next connection.
```